File: javascript/src/problems/move-zeroes.js
/**
 * 283. Move Zeroes [easy] https://leetcode.com/problems/move-zeroes/
 * 
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
const moveZeroes = nums => {
    let zeros = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 0) {
            zeros++;
        } else {
            nums[i-zeros] = nums[i];
        }
    }
    for (let i = nums.length - zeros; i < nums.length; i++) {
        nums[i] = 0;
    }
};

module.exports = { moveZeroes };

File: java/src/main/java/org/carlosbello/leetcode/problems/MoveZeroes.java
package org.carlosbello.leetcode.problems;

/**
 * 283. Move Zeroes [easy] https://leetcode.com/problems/move-zeroes/
 */
public class MoveZeroes {
    public void moveZeroes(int[] nums) {
        int zerosCount = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                zerosCount++;
            } else if (zerosCount > 0) {
                nums[i - zerosCount] = nums[i];
            }
        }
        while (zerosCount > 0) {
            nums[nums.length - zerosCount] = 0;
            zerosCount--;
        }
    }
}
